patches/0022-msdkenc-Align-16bit-for-progressive-mode.patch

```diff
@@ -0,0 +1,28 @@
+From 3d60df5973c21cd35fbb2a8d9797760161418ccb Mon Sep 17 00:00:00 2001
+From: "Ma, Mingyang" <mingyang.ma@intel.com>
+Date: Thu, 11 Aug 2022 10:36:22 +0800
+Subject: [PATCH] msdkenc: Align 16bit for progressive mode
+
+---
+ subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c | 5 ++++-
+ 1 file changed, 4 insertions(+), 1 deletion(-)
+
+diff --git a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c
+index 80b5c59a3d..e4b620b8e6 100644
+--- a/subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c
++++ b/subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c
+@@ -683,7 +683,10 @@ gst_msdkenc_init_encoder (GstMsdkEnc * thiz)
+   thiz->param.mfx.EncodedOrder = 0;     /* Take input frames in display order */
+ 
+   thiz->param.mfx.FrameInfo.Width = GST_ROUND_UP_16 (info->width);
+-  thiz->param.mfx.FrameInfo.Height = GST_ROUND_UP_32 (info->height);
++  if (info->interlace_mode != GST_VIDEO_INTERLACE_MODE_PROGRESSIVE)
```

Comment:
If interlace_mode is not progressive, we need to set PicStruct accordingly.

Reply:
> If interlace_mode is not progressive, we need to set PicStruct accordingly.

I know what you mean. But there is no method to set PicStruct by command line and maybe it is not the concern of the 
incorrect alignment. The previous version only provides progressive mode.
Should I fix all of these in one commit?

Reply:
> > If interlace_mode is not progressive, we need to set PicStruct accordingly.
> 
> I know what you mean. But there is no method to set PicStruct by command line and maybe it is not the concern of the incorrect alignment. The previous version only provides progressive mode. Should I fix all of these in one commit?

I don't think PicStruct should be set by cmd line, instead, it relates to interlace_mode which can be obtained from GstVideoInfo. Since the alignment for height relates to interlace_mode,  and interlace_mode impacts picture struct, so picture struct needs to be set accordingly.

Reply:
With @HeJunyan advice, we may not need to encode interlaced mode, so just add an assertion to interlace_mode, i.e. we only support progressive mode, and then directly set the alignment of Height to 16.
@xhaihao, any suggestion? Thanks.

Reply:
Is there any issue if setting the alignment of Height to 32 ?

Reply:
> Is there any issue if setting the alignment of Height to 32 ?

In different alignments, like the height of 720, the encoding stream of 32bit alignment is different from the 16bit

Reply:
> > Is there any issue if setting the alignment of Height to 32 ?

ffmpeg-qsv utilizes 16bit alignment for progressive (depending on version). What I am doing is trying to make the encoded streams of ffmpeg-qsv and gst-msdk are same

Reply:
If the coded stream doesn't violate the spec, I don't think we need to change the settings here.

Reply:
> If the coded stream doesn't violate the spec, I don't think we need to change the settings here.

The Height is not set correctly according to MediaSDK document. But I am not sure whether it's the spec you said. https://github.com/Intel-Media-SDK/MediaSDK/blob/master/doc/mediasdk-man.md#mfxFrameInfo:~:text=enumerator%20for%20details.-,Width%2C%20Height